oauthproxy.go

```diff
@@ -319,7 +319,7 @@ func (p *OAuthProxy) buildProxySubrouter(s *mux.Router) {
 	s.Path(signInPath).HandlerFunc(p.SignIn)
 	s.Path(signOutPath).HandlerFunc(p.SignOut)
 	s.Path(oauthStartPath).HandlerFunc(p.OAuthStart)
-	s.Path(oauthCallbackPath).HandlerFunc(p.OAuthCallback)
+	s.Path(oauthCallbackPath).Handler(p.sessionChain.ThenFunc(p.OAuthCallback))
```

Comment:
This is quite a major change to the way the project works. Previously we had always made sure that existing sessions didn't affect the attempts to re-authenticate, which seems like a valid use case to me.

What are the implications of this change?